internal/controller/nonadminbackup_controller.go

```diff
@@ -117,6 +117,14 @@ func (r *NonAdminBackupReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 			r.removeNabFinalizerUponVeleroBackupDeletion,
 		}
 
+	case function.CheckLabelAnnotationValueIsValid(nab.Labels, constant.NabSyncLabel):
+		logger.V(1).Info("Executing nab sync path")
+		reconcileSteps = []nonAdminBackupReconcileStepFunction{
+			r.setBackupUUIDInStatus,
+			r.setFinalizerOnNonAdminBackup,
+			r.createVeleroBackupAndSyncWithNonAdminBackup,
```

Comment:
I suppose the createVeleroBackup will not really create velero backup and will only sync it back to the synced NAB object? Velero Backup is the one triggering sync not vice-versa.

Reply:
yes, when synced nab reaches that step, it will fetch already created velero backup with its UUID instead of creating new one